### PR TITLE
Refactor DisplayManager, add UI elements

### LINKED ISSUE #18: Add UI
As the title says, something that tells the player its current status, combat log, pause screen, etc

### PR DESCRIPTION
**Pull request by Elkasitu**
With this commit the DisplayManager class gets a mini-refactoring to
make everything cleaner in order to fit the needs of rendering the game
UI.

This commit also adds a UI by default which is composed (for now) of
stat bars (HP and MP) and by a game log (player-readable messages), this
also comes with the possibility of painlessly adding new bars and adding
messages upon certain actions (like player attack)

Closes #18

### CODE REVIEW

File: display_manager.py [hunk 1]
```diff
@@ -27,14 +27,27 @@ class DisplayManager(metaclass=Singleton):
...
+    MSG_X = BAR_WIDTH + 2
+    MSG_WIDTH = SCREEN_WIDTH - BAR_WIDTH - 2
+    MSG_HEIGHT = PANEL_HEIGHT - 1
+
+    game_msgs = []
+
+    # TODO: give consoles a better name
     tdl.set_font('lucida10x10_gs_tc.png', greyscale=True, altLayout=True)
-    console = tdl.Console(SCREEN_WIDTH, SCREEN_HEIGHT)
+    console = tdl.Console(SCREEN_WIDTH, SCREEN_HEIGHT-4)
```
Comment (monzo94) [inline display_manager.py]: Where does this 4 come from? Why not use `PANEL_HEIGHT` or make it a new constant?

File: display_manager.py [hunk 2]
```diff
@@ -44,14 +57,105 @@ def gen_map(cls, player):
...
+
+        To general gameplay messages:
+            * Exp gained
+            * Loot pickups
+            * Actions performed / to perform
+
+        To NPC interactions:
+            * Dialog
+            * Trading
+
```
Comment (monzo94) [inline display_manager.py]: Do we need that many examples? It's pretty clear without them already.

File: display_manager.py [hunk 3]
```diff
@@ -44,14 +57,105 @@ def gen_map(cls, player):
...
+        bar_width = int(float(val) / maxi * total_w)
+        cls.panel.draw_rect(x, y, total_w, 1, None, bg=bg_color)
+
+        if bar_width > 0:
+            cls.panel.draw_rect(x, y, bar_width, 1, None, bg=fg_color)
+
+        # FIXME: make val be an int or a properly truncated float, don't coerce
+        text = f"{name}: {int(val)}/{maxi}"
+        x_centered = x + (total_w - len(text))//2
+        cls.panel.draw_str(x_centered, y, text, fg=Colors.WHITE, bg=None)
```
Comment (monzo94) [inline display_manager.py]: You could add an optional text color argument.

File: display_manager.py (hunk 1 above)
Comment (Elkasitu) [reply display_manager.py]: whoops

File: display_manager.py (hunk 3 above)
Comment (Elkasitu) [reply display_manager.py]: good idea

**Pull request merged**